typing_inspect/typing_inspect.py

```diff
@@ -0,0 +1,248 @@
+        is_tuple_type(tuple) == False
+        is_tuple_type(Tuple) == True
+        is_tuple_type(Tuple[str, int]) == True
+        class MyClass(Tuple[str, int]):
+            ...
+        is_tuple_type(MyClass) == True
+
+    For more general tests use issubclass(..., tuple), for more precise test use::
+
+        get_origin(tp) is Tuple
+    """
+
+    return type(tp) is type(Tuple)
+
+
+def is_union_type(tp):
+    """Test if the type is a union type. Examples::
+
+        is_union_type(int) == False
+        is_union_type(Union) == True
+        is_union_type(Union[int, int]) == False
+        is_union_type(Union[T, int]) == True
+    """
+
+    return type(tp) is type(Union)
+
+
+def is_typevar(tp):
+    """Test if the type represents a type variable. Examples::
+
+        is_typevar(int) == False
+        is_typevar(T) == True
+        is_typevar(Union[T, int]) == False
+    """
+
+    return type(tp) is TypeVar
+
+
+def is_classvar(tp):
+    """Test if the type represents a class variable. Examples::
+
+        is_classvar(int) == False
+        is_classvar(ClassVar) == True
+        is_classvar(ClassVar[int]) == True
+        is_classvar(ClassVar[List[T]]) == True
+    """
+
+    return type(tp) is type(ClassVar)
+
+
+def get_origin(tp):
+    """Get the unsubscripted version of a type. Supports generic types, Union,
+    Callable, and Tuple. Returns None for unsupported types. Examples::
+
+        get_origin(int) == None
+        get_origin(ClassVar[int]) == None
+        get_origin(Generic) == Generic
+        get_origin(Generic[T]) == Generic
+        get_origin(Union[T, int]) == Union
+        get_origin(List[Tuple[T, T]][int]) == List
+    """
+
+    if isinstance(tp, type(Generic)):
+        return _gorg(tp)
+    if is_union_type(tp):
+        return Union
+
+    return None
+
+
+def get_parameters(tp):
+    """Return type parameters of a parameterizable type as a tuple
+    in lexicographic order. Parameterizable types are generic types,
+    unions, tuple types and callable types. Examples::
+
+        get_parameters(int) == ()
+        get_parameters(Generic) == ()
+        get_parameters(Union) == ()
+        get_parameters(List[int]) == ()
+
+        get_parameters(Generic[T]) == (T,)
+        get_parameters(Tuple[List[T], List[S_co]]) == (T, S_co)
+        get_parameters(Union[S_co, Tuple[T, T]][int, U]) == (U,)
+        get_parameters(Mapping[T, Tuple[S_co, T]]) == (T, S_co)
+    """
+
+    if (
+        is_generic_type(tp) or is_union_type(tp) or
+        is_callable_type(tp) or is_tuple_type(tp)
+    ):
+        return tp.__parameters__ if tp.__parameters__ is not None else ()
+    return ()
+
+
+def get_last_args(tp):
+    """Get last arguments of (multiply) subscripted type. Examples::
+
+        get_last_args(int) == ()
+        get_last_args(Union) == ()
+        get_last_args(ClassVar[int]) == (int,)
+        get_last_args(Union[T, int]) == (T, int)
+        get_last_args(Iterable[Tuple[T, S]][int, T]) == (int, T)
```

Comment:
Could you add an example here for Callable? It's not obvious to me what this will return for something like `Callable[[int], str]`.